drivers/axi_core/spi_engine/spi_engine.c

```diff
@@ -0,0 +1,958 @@
+			spi_engine_set_cs(desc, true);
+		} else if(parameter == 0x00) {
+			dev_dbg(desc, "CS LOW");
+			/* Set the CS LOW */
+			spi_engine_set_cs(desc, false);
+		}
+		break;
+
+	/* The SYNC and SLEEP commands got the same value but different
+	modifier */
+	case SPI_ENGINE_INST_SYNC_SLEEP:
+		/* SYNC instruction */
+		if(modifier == 0x00) {
+			dev_dbg(desc, "Sync ID : 0x%X", parameter);
+			spi_engine_write_cmd_reg(desc_extra, cmd);
+		} else if(modifier == 0x01) {
+			spi_gen_sleep_ns(desc, parameter);
+		}
+		break;
+	case SPI_ENGINE_INST_CONFIG:
+		switch(modifier) {
+		case 0x00:
+			dev_dbg(desc, "Prescaler set to 0x%X", parameter);
+			break;
+		case 0x01:
+			dev_dbg(desc, "SPI mode set to 0x%X", parameter);
+			break;
+		case 0x02:
+			dev_dbg(desc, "Transfer length set to 0x%X", parameter);
+			break;
+		}
+		spi_engine_write_cmd_reg(desc_extra, cmd);
+
+		break;
+
+	default:
+		dev_err(desc, "Invalid command");
+
+		return FAILURE;
+		break;
+	}
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Prepare the command queue before sending it to the engine
+ *
+ * @param desc Decriptor containing SPI interface parameters
+ * @param msg Structure used to store the transfer messages
+ * @return int32_t This function allways returns SUCCESS
+ */
+int32_t spi_engine_compile_message(struct spi_desc *desc,
+				   struct spi_engine_msg *msg)
+{
+	struct spi_engine_desc	*desc_extra;
+
+	desc_extra = desc->extra;
+
+	/* Set the data transfer length */
+	spi_engine_queue_append_cmd(&msg->cmds,
+				    SPI_ENGINE_CMD_CONFIG(
+					SPI_ENGINE_CMD_DATA_TRANSFER_LEN,
+					desc_extra->data_width));
+	/*
+	 * Configure the spi mode :
+	 *	- 3 wire
+	 *	- CPOL
+	 *	- CPHA
+	 */
+	spi_engine_queue_append_cmd(&msg->cmds,
+				    SPI_ENGINE_CMD_CONFIG(
+					SPI_ENGINE_CMD_REG_CONFIG,
+					desc->mode));
+
+	/* Configure the prescaler */
+	spi_engine_queue_append_cmd(&msg->cmds,
+				    SPI_ENGINE_CMD_CONFIG(
+					SPI_ENGINE_CMD_REG_CLK_DIV,
+					desc_extra->clk_div));
+
+	/* Add a sync command to signal that the transfer has finished */
+	spi_engine_queue_add_cmd(&msg->cmds, SPI_ENGINE_CMD_SYNC(_sync_id));
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Initiate a spi transfer
+ *
+ * @param desc Decriptor containing SPI interface parameters
+ * @param msg Structure used to store the transfer messages
+ * @return int32_t - SUCCESS if the transfer finished
+ *		   - FAILURE if the memory allocation failed
+ */
+int32_t spi_engine_transfer_message(struct spi_desc *desc,
+				    struct spi_engine_msg *msg)
+{
+	uint32_t		i;
+	uint32_t		data;
+	uint32_t		sync_id;
+	bool 			offload_en;
+	struct spi_engine_desc	*desc_extra;
+
+	desc_extra = desc->extra;
+
+	spi_engine_compile_message(desc, msg);
+
+	offload_en = (desc_extra->offload_config & OFFLOAD_TX_EN) |
+		     (desc_extra->offload_config & OFFLOAD_RX_EN);
+
+	/* Write the command fifo buffer */
+	do {
+		spi_engine_queue_lifo_get_cmd(&msg->cmds, &data);
+		spi_engine_write_cmd(desc, data);
+		usleep(1000);
```

Comment:
This looks like a hack to make it work. Is this really needed ? If yes, please document why is it needed.

Reply:
This was forgotten after some tests and will be removed.